spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/StreamListenerMethodUtils.java

```diff
@@ -0,0 +1,151 @@
+			String methodAnnotatedInboundName, String methodAnnotatedOutboundName, boolean isDeclarative) {
+		int methodArgumentsLength = method.getParameterTypes().length;
+		if (!isDeclarative) {
+			Assert.isTrue(inputAnnotationCount == 0 && outputAnnotationCount == 0, INVALID_MESSAGE_HANDLER_METHOD_PARAMS);
+		}
+		if (StringUtils.hasText(methodAnnotatedInboundName) && StringUtils.hasText(methodAnnotatedOutboundName)) {
+			Assert.isTrue(inputAnnotationCount == 0 && outputAnnotationCount == 0, INVALID_INPUT_OUTPUT_METHOD_PARAMETERS);
+		}
+		if (StringUtils.hasText(methodAnnotatedInboundName)) {
+			Assert.isTrue(inputAnnotationCount == 0, INVALID_INPUT_VALUES);
+			Assert.isTrue(outputAnnotationCount == 0, INVALID_INPUT_VALUE_WITH_OUTPUT_METHOD_PARAM);
+		}
+		else {
+			Assert.isTrue(inputAnnotationCount >= 1, NO_INPUT_DESTINATION);
+		}
+		if (StringUtils.hasText(methodAnnotatedOutboundName)) {
+			Assert.isTrue(outputAnnotationCount == 0, INVALID_OUTPUT_VALUES);
+		}
+		if (isDeclarative) {
+			for (int parameterIndex = 0; parameterIndex < methodArgumentsLength; parameterIndex++) {
+				MethodParameter methodParameter = MethodParameter.forMethodOrConstructor(method, parameterIndex);
+				if (methodParameter.hasParameterAnnotation(Input.class)) {
+					String inboundName = (String) AnnotationUtils.getValue(methodParameter.getParameterAnnotation(Input.class));
+					Assert.isTrue(StringUtils.hasText(inboundName), INVALID_INBOUND_NAME);
+				}
+				if (methodParameter.hasParameterAnnotation(Output.class)) {
+					String outboundName = (String) AnnotationUtils.getValue(methodParameter.getParameterAnnotation(Output.class));
+					Assert.isTrue(StringUtils.hasText(outboundName), INVALID_OUTBOUND_NAME);
+				}
+			}
+			if (methodArgumentsLength > 1){
+				Assert.isTrue(inputAnnotationCount + outputAnnotationCount == methodArgumentsLength, INVALID_DECLARATIVE_METHOD_PARAMETERS);
+			}
+		}
+	}
+
+	protected static void assertStreamListenerMessageHandlerMethod(Method method) {
+		int methodArgumentsLength = method.getParameterTypes().length;
+		if (methodArgumentsLength > 1) {
+			int numAnnotatedMethodParameters = 0;
+			int numPayloadAnnotations = 0;
+			for (int parameterIndex = 0; parameterIndex < methodArgumentsLength; parameterIndex++) {
+				MethodParameter methodParameter = MethodParameter.forMethodOrConstructor(method, parameterIndex);
+				if (methodParameter.hasParameterAnnotations()) {
+					numAnnotatedMethodParameters++;
+				}
+				if (methodParameter.hasParameterAnnotation(Payload.class)) {
+					numPayloadAnnotations++;
+				}
+			}
+			if (numPayloadAnnotations > 0) {
+				Assert.isTrue(methodArgumentsLength == numAnnotatedMethodParameters && numPayloadAnnotations <= 1,
+						AMBIGUOUS_MESSAGE_HANDLER_METHOD_ARGUMENTS);
+			}
+		}
+	}
+
+	protected static String getInboundElementNameFromMethod(StreamListener streamListener) {
```

Comment:
We can inline this as it is simply getting the value from `StreamListener` - and returning `null` if empty.